thirdeye-server/src/main/java/ai/startree/thirdeye/resources/NamespaceConfigurationResource.java

```diff
@@ -0,0 +1,145 @@
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+@Tag(name = "Namespace Configuration")
+@SecurityRequirement(name="oauth")
+@SecurityRequirement(name = Constants.NAMESPACE_SECURITY)
+@OpenAPIDefinition(security = {
+    @SecurityRequirement(name = "oauth"),
+    @SecurityRequirement(name = Constants.NAMESPACE_SECURITY)
+})
+@SecurityScheme(name = "oauth", type = SecuritySchemeType.APIKEY, in = SecuritySchemeIn.HEADER, paramName = HttpHeaders.AUTHORIZATION)
+@SecurityScheme(name = Constants.NAMESPACE_SECURITY, type = SecuritySchemeType.APIKEY, in = SecuritySchemeIn.HEADER, paramName = Constants.NAMESPACE_HTTP_HEADER)
+@Singleton
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class NamespaceConfigurationResource extends CrudResource<NamespaceConfigurationApi,
+    NamespaceConfigurationDTO> {
+
+  private final NamespaceConfigurationService namespaceConfigurationService;
+
+  @Inject
+  public NamespaceConfigurationResource(
+      final NamespaceConfigurationService namespaceConfigurationService) {
+    super(namespaceConfigurationService);
+    this.namespaceConfigurationService = namespaceConfigurationService;
+  }
+
+  @Override
+  @GET
+  @Timed(percentiles = {0.5, 0.75, 0.90, 0.95, 0.98, 0.99, 0.999})
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response list(
+      @Parameter(hidden = true) @Auth ThirdEyeServerPrincipal principal,
+      @Context UriInfo uriInfo
+  ) {
+    return respondOk(namespaceConfigurationService.getNamespaceConfiguration(principal));
+  }
+
+  // Overridden to disable endpoint
+  @Override
+  @GET
+  @Path("{id}")
+  @Operation(summary = "", hidden = true)
+  @Timed(percentiles = {0.5, 0.75, 0.90, 0.95, 0.98, 0.99, 0.999})
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response get(
+      @Parameter(hidden = true) @Auth ThirdEyeServerPrincipal principal,
+      @PathParam("id") Long id) {
+    throw new UnsupportedOperationException();
+  }
+
+  // Overridden to disable endpoint
+  @Override
+  @POST
+  @Operation(summary = "", hidden = true)
+  @Timed(percentiles = {0.5, 0.75, 0.90, 0.95, 0.98, 0.99, 0.999})
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response createMultiple(
+      @Parameter(hidden = true) @Auth final ThirdEyeServerPrincipal principal,
+      final List<NamespaceConfigurationApi> list) {
+    throw new UnsupportedOperationException();
+  }
+
+  // Overridden to disable endpoint
+  @Override
+  @PUT
+  @Operation(summary = "", hidden = true)
+  @Timed(percentiles = {0.5, 0.75, 0.90, 0.95, 0.98, 0.99, 0.999})
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response editMultiple(
+      @Parameter(hidden = true) @Auth final ThirdEyeServerPrincipal principal,
+      final List<NamespaceConfigurationApi> list) {
+    throw new UnsupportedOperationException();
```

Comment:
we do need a PUT to edit right? see comment above

Reply:
Yes, i was planning to add it later. but it's a small change. Adding